internal/mode/static/nginx/agent/grpc/connections.go

```diff
@@ -2,49 +2,94 @@ package grpc
 
 import (
 	"sync"
+
+	"k8s.io/apimachinery/pkg/types"
 )
 
-// ConnectionsTracker keeps track of all connections between the control plane and nginx agents.
-type ConnectionsTracker struct {
-	// connections contains a map of all IP addresses that have connected and their associated pod names.
-	// TODO(sberman): we'll likely need to create a channel for each connection that can be stored in this map.
-	// Then the Subscription listens on the channel for its connection, while the nginxUpdater sends the config
-	// for the pod over that channel.
-	connections map[string]string
+//go:generate go run github.com/maxbrunsfeld/counterfeiter/v6 -generate
+
+//counterfeiter:generate . ConnectionsTracker
+
+// ConnectionsTracker defines an interface to track all connections between the control plane
+// and nginx agents.
+type ConnectionsTracker interface {
+	Track(key string, conn Connection)
+	GetConnection(key string) Connection
+	ConnectionIsReady(key string) (Connection, bool)
```

Comment:
Do we need both GetConnection and ConnectionIsReady? Why not just add a method to Connection called Ready?

Reply:
GetConnetion returns the connection regardless of state, so I do think both functions are necessary. But I'll rename to Ready().

Reply:
Couldn't the caller check if the connection is ready? 

```
conn := GetConnection(connKey)
if conn.Ready() { // do something}
```

Reply:
Oh I see. Yeah, I could do that too.